### PR TITLE
feat(billing): add billing command

### PR DESCRIPTION
**Pull request by Monitob**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/master/CHANGELOG.md):

```release-note

```

### CODE REVIEW

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		{
+			Name:       "file-name",
+			Short:      `Wanted file name`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+		},
+		{
+			Name:       "file-path",
+			Short:      `Wanted file locality`,
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:54]:
```suggestion
			Short:      `Wanted file path`,
```

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			Positional: false,
+			EnumValues: []string{"pdf"},
+		},
+		{
+			Name:       "file-name",
+			Short:      `Wanted file name`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+		},
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:51]: IMO we should have path & name in a single flag (--output?)

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	// set file name
+	fileName := resp.Name
+	if len(argsDownload.FileName) > 0 {
+		fileName = fileNameWithoutExtTrimSuffix(argsDownload.FileName)
+	}
+
+	// check content-type
+	switch resp.ContentType {
+	case "application/pdf":
+		fileName = fmt.Sprintf("%s.pdf", fileName)
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:95]: IMO we leave the user choose the extension. What do you think @Codelax ?

```suggestion
		fileName = fmt.Sprintf("%s", fileName)
```

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go (hunk 3 above)
Comment (Monitob) [reply internal/namespaces/billing/v2alpha1/custom_billing_download.go]: Only the supported extensions for now. Maybe we discuss this in another ticket.

File: cmd/scw/testdata/test-all-usage-billing-invoice-download-usage.golden [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Download billing resources.
+
+USAGE:
+  scw billing invoice download [arg=value ...]
+
+ARGS:
+  invoice-id                                          Invoice ID
+  [file-type]                                         Wanted file type (pdf)
+  [file-path]                                         Wanted file locality
+  [file-type=File extension `.pdf` as default]        Wanted file extension
```
Comment (kindermoumoute) [inline cmd/scw/testdata/test-all-usage-billing-invoice-download-usage.golden:12]: doc in the wrong place + duplicate `file-type`

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+		},
+		{
+			Name:       "file-type",
+			Short:      `Wanted file extension`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+			Default: func(ctx context.Context) (value string, doc string) {
+				return billing.DownloadInvoiceRequestFileTypePdf.String(), "File extension `.pdf` as default"
+			},
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:68]: This is not how to use Default func

```suggestion
			Default: core.DefaultValueSetter(billing.DownloadInvoiceRequestFileTypePdf.String()),
```

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+				askPrompt = true
+			}
+		}
+
+		if askPrompt {
+			_, _ = interactive.PrintlnWithoutIndent(`
+					Current file exist is located at ` + terminal.Style(args.FilePath, color.Faint))
+			overrideFile, err := interactive.PromptBoolWithConfig(&interactive.PromptBoolConfig{
+				Prompt:       fmt.Sprintf("Do you want to override the current file: %s ?", file),
+				DefaultValue: true,
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:141]: why true by default?

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+			}
+		}
+
+		entries, err := os.ReadDir(dir)
+		if err != nil {
+			return err
+		}
+
+		if args.ForceReplace {
+			return nil
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:115]: I don't understand why it returns here

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+			Deprecated: false,
+			Positional: false,
+		},
+		{
+			Name:       "file-path",
+			Short:      `Wanted file path`,
+			Required:   false,
+			Deprecated: false,
+			Positional: false,
+			Default:    core.DefaultValueSetter("/tmp/"),
```
Comment (kindermoumoute) [inline internal/namespaces/billing/v2alpha1/custom_billing_download.go:51]: Can we use local directory by default? "tmp" does not exist on windows
```suggestion
			Default:    core.DefaultValueSetter("./"),
```

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: internal/namespaces/billing/v2alpha1/custom_billing_download.go (hunk 8 above)
Comment (Codelax) [reply internal/namespaces/billing/v2alpha1/custom_billing_download.go:51]: If what you want is a temporary directory, have a look at `core.ExtractCacheDir`

**Pull request merged**